afkak/test/test_consumer.py

```diff
@@ -785,16 +785,23 @@ def test_consumer_error_during_offset(self):
     def test_consumer_offset_out_of_range_error(self):
         topic = 'offset_out_of_range_error'
         part = 911
+        offset = 10000
         reqs_ds = [Deferred(), Deferred()]
         mockclient = Mock()
         mockclient.send_offset_request.side_effect = reqs_ds
```

Comment:
You're mocking the `send_offset_request` method on the client, but since now you're starting the consumer with a specific offset, `send_offset_request` shouldn't be called (at least not at first). Instead, you need to set the `send_fetch_request` side_effect to those deferreds so you can errback(f) them like you're doing below.  Then depending on how deep you want to do the mocking, you can either just check that `client.send_offset_request` was called with the proper request, or you can setup the side effects for that mock as well and make sure the consumer makes the proper fetch request with the proper offset.  I'd probably just check that the right `send_offset_request` was made after you `errback()` the send_fetch_request.

Reply:
Thank you so much @rthille  for the hints, which is appreciated!